Improve package import error message.

Suppose we have this structure:
```
Django view Injector
  Package attribute
    Service Injector
      Package attribute
        Query function
```
For example, a module containing this query function has an import error. In this case, the user will see an ugly unclear error:
```python
tests/test_upload_attachment.py:30: in test_attachment_details
    def test_attachment_details():
.venv/lib/python3.7/site-packages/pytest_bdd/scenario.py:195: in _execute_scenario
    _execute_step_function(request, scenario, step, step_func)
.venv/lib/python3.7/site-packages/pytest_bdd/scenario.py:136: in _execute_step_function
    step_func(**kwargs)
tests/test_upload_attachment.py:54: in uploaded_image
    {'type': 'image', 'image': jpeg_image},
tests/fixtures/client.py:43: in __call__
    response = getattr(self.api, self.verb)(**kwargs)
.venv/lib/python3.7/site-packages/rest_framework/test.py:300: in post
    path, data=data, format=format, content_type=content_type, **extra)
.venv/lib/python3.7/site-packages/rest_framework/test.py:213: in post
    return self.generic('POST', path, data, content_type, **extra)
.venv/lib/python3.7/site-packages/rest_framework/test.py:238: in generic
    method, path, data, content_type, secure, **extra)
.venv/lib/python3.7/site-packages/django/test/client.py:422: in generic
    return self.request(**r)
.venv/lib/python3.7/site-packages/rest_framework/test.py:289: in request
    return super(APIClient, self).request(**kwargs)
.venv/lib/python3.7/site-packages/rest_framework/test.py:241: in request
    request = super(APIRequestFactory, self).request(**kwargs)
.venv/lib/python3.7/site-packages/django/test/client.py:503: in request
    raise exc_value
.venv/lib/python3.7/site-packages/django/core/handlers/exception.py:34: in inner
    response = get_response(request)
.venv/lib/python3.7/site-packages/django/core/handlers/base.py:115: in _get_response
    response = self.process_exception_by_middleware(e, request)
.venv/lib/python3.7/site-packages/django/core/handlers/base.py:113: in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
.venv/lib/python3.7/site-packages/django/views/decorators/csrf.py:54: in wrapped_view
    return view_func(*args, **kwargs)
.venv/lib/python3.7/site-packages/rest_framework/viewsets.py:116: in view
    return self.dispatch(request, *args, **kwargs)
.venv/lib/python3.7/site-packages/rest_framework/views.py:495: in dispatch
    response = self.handle_exception(exc)
.venv/lib/python3.7/site-packages/rest_framework/views.py:455: in handle_exception
    self.raise_uncaught_exception(exc)
.venv/lib/python3.7/site-packages/rest_framework/views.py:492: in dispatch
    response = handler(request, *args, **kwargs)
.venv/lib/python3.7/site-packages/rest_framework/mixins.py:21: in create
    self.perform_create(serializer)
.venv/src/dependencies/src/dependencies/contrib/_rest_framework.py:180: in __method
    cb(argument, getattr(ns, method)())
.venv/src/dependencies/src/dependencies/_injector.py:77: in __getattr__
    cache[current_attr] = attribute(**kwargs)
.venv/src/dependencies/src/dependencies/_attributes.py:18: in __call__
    result = getattr(result, attr)
.venv/src/dependencies/src/dependencies/_injector.py:77: in __getattr__
    cache[current_attr] = attribute(**kwargs)
.venv/src/dependencies/src/dependencies/_package.py:44: in __call__
    result = getattr(result, attr)
E   AttributeError: module 'server.models' has no attribute 'attachment'
```
The main problem here **we can't even understand what attributes in what Injector subclass cause this error** because it's location is hidden by the import error.

--- a/src/_dependencies/objects/package.py
+++ b/src/_dependencies/objects/package.py
@@ -1,4 +1,5 @@
 from importlib import import_module
+from inspect import ismodule
 
 from _dependencies.objects.attributes import _Attributes
 
@@ -39,10 +40,11 @@ def _import_module(module, attrs):
     index = 0
     for attr in attrs:
         index += 1
-        try:
-            module += "." + attr
-            result = import_module(module)
-        except ImportError:
-            result = getattr(result, attr)
-            break
+        if hasattr(result, attr):
+            attribute = getattr(result, attr)
+            if not ismodule(attribute):
+                result = attribute
+                break
+        module += "." + attr
+        result = import_module(module)
     return result, attrs[index:]

--- a/src/_dependencies/objects/this.py
+++ b/src/_dependencies/objects/this.py
@@ -56,9 +56,8 @@ def _get_attribute(instance, name):
         return getattr(instance, name)
     except DependencyError:
         if name == "__parent__":
-            raise DependencyError(
-                "You tried to shift this more times than Injector has levels"
-            )
+            message = "You tried to shift this more times than Injector has levels"
+            raise DependencyError(message) from None
         else:
             raise
 

--- a/src/_dependencies/resolve.py
+++ b/src/_dependencies/resolve.py
@@ -15,7 +15,7 @@ class _Resolver:
         except RecursionError:
             message = _Trace(self.state)
             message.add("Circle error found in definition of the dependency graph")
-            raise DependencyError(message)
+            raise DependencyError(message) from None
 
     def find(self):
         while self.attrname not in self.state.cache:
@@ -44,7 +44,7 @@ class _Resolver:
         except DependencyError as error:
             message = _Trace(self.state)
             message.add(error)
-            raise DependencyError(message)
+            raise DependencyError(message) from None
 
     def match(self, args):
         for arg, have_default in args.items():  # pragma: no branch

--- a/tests/helpers/examples/__init__.py
+++ b/tests/helpers/examples/__init__.py
@@ -1,0 +1,1 @@
+init_variable = 1

--- a/tests/helpers/examples/import_error.py
+++ b/tests/helpers/examples/import_error.py
@@ -1,0 +1,1 @@
+from astral import Vision  # noqa: F401

--- a/tests/test_objects_package.py
+++ b/tests/test_objects_package.py
@@ -5,6 +5,7 @@ import pytest
 
 from dependencies import Injector
 from dependencies import Package
+from dependencies import this
 from dependencies.exceptions import DependencyError
 from helpers import CodeCollector
 
@@ -128,6 +129,12 @@ def _xqImNk0p0kun():
 def _d1H5hbDhnoVL():
     submodule = Package("examples.submodule")
     return submodule.variable
+
+
+@variable_variant
+def _ocFk5G1OWGty():
+    examples = Package("examples")
+    return examples.init_variable
 
 
 provide_function = CodeCollector()
@@ -428,3 +435,71 @@ def _dmldmoXCFIBG():
 def _pRX4SSAbG8iO():
     examples = Package("examples")
     return Injector(foo=examples.self_pointer.Container.foo)
+
+
+import_error = CodeCollector()
+nested_variant = CodeCollector("nested")
+error_variant = CodeCollector("error")
+
+
+@import_error.parametrize
+@nested_variant.parametrize
+@error_variant.parametrize
+def test_handle_import_error(code, nested, error):
+    """Import time errors should be propagated.
+
+    Import error raised by poorly written project module should not be hidden by Package
+    implementation.
+
+    """
+
+    class Root:
+        def __init__(self, result):
+            raise RuntimeError
+
+    Container = code(Root, nested(error()))
+
+    with pytest.raises(ModuleNotFoundError) as exc_info:
+        Container.root
+
+    assert str(exc_info.value) == "No module named 'astral'"
+
+
+@import_error
+def _yjihTbbUJ3rS(Root, Nested):
+    class Container(Injector):
+        root = Root
+        result = this.nested.foo
+        nested = Nested
+
+    return Container
+
+
+@import_error
+def _xtLbizL0vuJr(Root, Nested):
+    return Injector(root=Root, result=this.nested.foo, nested=Nested)
+
+
+@nested_variant
+def _cdYU3Mxu5W0R(error):
+    class Nested(Injector):
+        foo = error
+
+    return Nested
+
+
+@nested_variant
+def _q8vFClchGiWW(error):
+    return Injector(foo=error)
+
+
+@error_variant
+def _cXjECOua1K75():
+    examples = Package("examples")
+    return examples.import_error.Vision
+
+
+@error_variant
+def _jNELGA2KFLA7():
+    submodule = Package("examples.import_error")
+    return submodule.Vision

